src/ui/features/onboarding/Onboarding.tsx

```diff
@@ -66,12 +86,14 @@ export function Onboarding() {
         path="/onboarding/session-expired"
         element={
           <PageLayout>
-            <SessionExpired onSubmit={() => navigate('/onboarding')} />
+            <SessionExpired
+              onRestart={
+                existingUser ? () => navigate('/onboarding') : undefined
```

Comment:
Why `undefined`? Can't navigation restart is no user had been created?

Reply:
It should be the other way around. Fixed & moved this logic to the `SessionExpired` component to make it more readable ✅